### ISSUE TITLE
Issue with optimizer data type

### ISSUE THREAD

**Issue by kritiyer** [jakeret/tf_unet#180]
I am trying to use the template from the usage docs to train a unet on my own data. I am able to successfully load my data using image_util.ImageDataProvider and initialize a unet with unet.Unet(). When I run:

`trainer= unet.Trainer(net, batch_size=10, optimizer='adam'`

I receive an error message that states:
```
InvalidArgumentError (see above for traceback): Cannot parse tensor from proto: dtype: DT_FLOAT
tensor_shape {
  dim {
    size: 3
  }
  dim {
    size: 3
  }
  dim {
    size: 8192
  }
  dim {
    size: 8192
  }
}
float_val: 0

	 [[Node: Variable_29/Adam/Initializer/zeros = Const[_class=["loc:@Variable_29"], dtype=DT_FLOAT, value=<Invalid TensorProto: dtype: DT_FLOAT tensor_shape { dim { size: 3 } dim { size: 3 } dim { size: 8192 } dim { size: 8192 } } float_val: 0>, _device="/job:localhost/replica:0/task:0/device:CPU:0"]()]]
```

I tried changing the optimizer to 'momentum' and received the same error message but with 'Node: Variable_29/Momentum/Initializer/zeros' instead. I'm not sure why the dim size variables say 3 and 8192 since my images are grayscale (1 channel) and 572x572 pixels. Do you know why the optimizer initializer is throwing these errors?

**Comment by jakeret** [comment]
Have you checked what the ImageDataProvider returns when you call `data_provider(1)`. Does the output make sense?

**Comment by kritiyer** [comment]
Yes, I checked ImageDataProvider and data_provider(1) returned the proper output shapes and types. I resolved the issue; it was actually an error due to insufficient memory. When I decreased the batch size and features_root parameters, the unet began training as expected.
